test/e2e/proxy.go

```diff
@@ -118,12 +122,31 @@ func proxyContext(version string) {
 		// Try proxying through the service and directly to through the pod.
 		svcPrefix := prefix + "/proxy/namespaces/" + f.Namespace.Name + "/services/" + service.Name
 		podPrefix := prefix + "/proxy/namespaces/" + f.Namespace.Name + "/pods/" + pods[0].Name
+		namespacedPodProxyURL := func(portAndSchemeSuffix string) string {
+			return prefix + "/namespaces/" + f.Namespace.Name + "/pods/" + pods[0].Name + portAndSchemeSuffix + "/proxy"
+		}
 		expectations := map[string]string{
 			svcPrefix + ":portname1/": "foo",
 			svcPrefix + ":portname2/": "bar",
 			podPrefix + ":80/":        `<a href="` + podPrefix + `:80/rewriteme">test</a>`,
 			podPrefix + ":160/":       "foo",
 			podPrefix + ":162/":       "bar",
+
+			podPrefix + ":80:http/":  `<a href="` + podPrefix + `:80:http/rewriteme">test</a>`,
```

Comment:
agree... my first pass did `scheme:name:port`, but I didn't like the meaning of part[0] changing depending on how many parts there were. I could be talked into switch back, though